test/unit/builtins/object.spec.ts

```diff
@@ -222,22 +222,42 @@ describe("delete from object", () => {
     test("delete from object", () => {
         util.testFunction`
             const obj = { foo: "bar", bar: "baz" };
-            delete obj["foo"];
-            return obj;
-        `
-            .setTsHeader("declare function setmetatable<T extends object>(this: void, table: T, metatable: any): T;")
-            .expectToEqual({ bar: "baz" });
+            return [delete obj["foo"], obj];
+        `.expectToMatchJsResult();
+    });
+
+    test("delete nonexistent property", () => {
+        util.testFunction`
+            const obj = {} as any
+            return [delete obj["foo"], obj];
+        `.expectToMatchJsResult();
     });
 
     // https://github.com/TypeScriptToLua/TypeScriptToLua/issues/993
     test("delete from object with metatable", () => {
         util.testFunction`
         const obj = { foo: "bar", bar: "baz" };
         setmetatable(obj, {});
-        delete obj["foo"];
-        return obj;
+        return [delete obj["foo"], obj];
     `
             .setTsHeader("declare function setmetatable<T extends object>(this: void, table: T, metatable: any): T;")
-            .expectToEqual({ bar: "baz" });
+            .expectToEqual([true, { bar: "baz" }]);
+    });
+
+    test("delete nonconfigurable own property", () => {
+        util.testFunction`
+
+        const obj = {};
+        Object.defineProperty(obj,"foo", {
+            configurable: false,
+            value: true
+        })
+        try {
+            delete obj["foo"]
+            return "deleted"
+        } catch (e) {
+            return e instanceof TypeError
```

Comment:
Looks like this branch is not actually hit in this test, could you add a test where the TypeError is thrown? Also indenting seems worng here, it should be increased by 1 indent on line 250 to 261, empty line 249 should be removed.

Reply:
The test _does_ throw TypeError (the return value is true), at least when I run on my computer. Did you recompile lualib, or maybe is strict mode off somehow?

I added strict mode explicitly to the test just in case.